libs/community/langchain_community/vectorstores/azuresearch.py

```diff
@@ -451,39 +413,30 @@ def vector_search_with_score(
         Returns:
             List of Documents most similar to the query and score for each
         """
-        from azure.search.documents.models import Vector
+
+        from azure.search.documents.models import VectorizedQuery
 
         results = self.client.search(
             search_text="",
-            vectors=[
-                Vector(
-                    value=np.array(self.embed_query(query), dtype=np.float32).tolist(),
-                    k=k,
+            vector_queries=[
+                VectorizedQuery(
+                    vector=np.array(self.embed_query(query), dtype=np.float32).tolist(),
+                    k_nearest_neighbors=k,
                     fields=FIELDS_CONTENT_VECTOR,
                 )
             ],
             filter=filters,
+            top=k,
         )
         # Convert results to Document objects
         docs = [
             (
                 Document(
                     page_content=result.pop(FIELDS_CONTENT),
-                    metadata={
-                        **(
-                            {FIELDS_ID: result.pop(FIELDS_ID)}
-                            if FIELDS_ID in result
-                            else {}
-                        ),
```

Comment:
hey @lz-chen @mattgotteiner, was there a specific reason we removed FIELDS_ID in this pr?